From 273ebecf7c676ece9b1143f6db7db86f411e08ab Mon Sep 17 00:00:00 2001
From: Eneas U de Queiroz <cotequeiroz@gmail.com>
Date: Tue, 27 Jul 2021 10:40:42 -0300
Subject: [PATCH 3/6] storage: send_beacon_reports: debug BSSID loop

See if we don't need to keep going after the initial mismatch between AP
and client BSSID.

Signed-off-by: Eneas U de Queiroz <cotequeiroz@gmail.com>
---
 src/storage/datastorage.c | 20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/src/storage/datastorage.c b/src/storage/datastorage.c
index 0d115a3..9e16382 100644
--- a/src/storage/datastorage.c
+++ b/src/storage/datastorage.c
@@ -396,10 +396,27 @@ void send_beacon_reports(ap *a, int id) {
 
     // Seach for BSSID
     client* i = *client_find_first_bc_entry(a->bssid_addr, dawn_mac_null, false);
+#ifndef DAWN_NO_OUTPUT
+    printf("Searching for BSSID=" MACSTR "\n", MAC2STR(a->bssid_addr.u8));
+    if (!i) printf("Result is NULL!\n");
+#endif
 
     // Go threw clients
-    while (i != NULL && mac_is_equal_bb(i->bssid_addr, a->bssid_addr)) {
+    while (i != NULL) {
+#ifndef DAWN_NO_OUTPUT
+        int mismatch = false;
+        printf("Checking AP_BSSID=" MACSTR " vs. STA_BSSID=" MACSTR "\n",
+               MAC2STR(a->bssid_addr.u8), MAC2STR(i->bssid_addr.u8));
+#endif
+        if (!mac_is_equal_bb(i->bssid_addr, a->bssid_addr)) {
+#ifndef DAWN_NO_OUTPUT
+            mismatch = true;
+#endif
+            goto next;
+        }
 #ifndef DAWN_NO_OUTPUT
+        if (mismatch)
+            printf("We should not even be here!\n");
         printf("Client " MACSTR ": rrm_enabled_capa=%02x: PASSIVE=%d, ACTIVE=%d, TABLE=%d\n",
             MAC2STR(i->client_addr.u8), i->rrm_enabled_capa,
             !!(i->rrm_enabled_capa & WLAN_RRM_CAPS_BEACON_REPORT_PASSIVE),
@@ -408,6 +425,7 @@ void send_beacon_reports(ap *a, int id) {
 #endif
         if (i->rrm_enabled_capa & dawn_metric.rrm_mode_mask)
             ubus_send_beacon_report(i, a, id);
+next:
         i = i->next_entry_bc;
     }
 
-- 
2.31.1

